src/manager_tests.cpp

```diff
@@ -21,182 +21,157 @@ struct Health
-	// However, it doesn't throw at all because it only "sees" health in EntitiesWith - after all the recursion.
-	CHECK_THROWS_AS(ecs.EntitiesWith(Health{}, Identity{}, AI{}), const ComponentNotRegisteredException);
-}
-
-TEST_CASE("manager REMOVE COMPONENT")
-{
-	ECS ecs;
+        // set up entity 1
+        Entity entity1 = ecs.CreateEntity();
 
-	ecs.RegisterComponent(Identity());
-	ecs.RegisterComponent(Health());
+        Identity ident{ "babs1" };
+        ecs.AddComponent(entity1, ident);
 
-	Identity e0Ident = Identity();
-	e0Ident.name = "babs2";
+        Health hp1{ 100, 44 };
+        ecs.AddComponent(entity1, hp1);
 
-	Identity e1Ident = Identity();
-	e1Ident.name = "babs1";
+        // set up entity 2
+        Entity entity2 = ecs.CreateEntity();
+        Health hp2{ 50, 22 };
+        ecs.AddComponent(entity2, hp2);
 
-	Health e0Health = Health();
-	e0Health.current = 100;
-	e0Health.max = 100;
+        // set up entity 3
+        Entity entity3 = ecs.CreateEntity();
 
-	Health e1Health = Health();
-	e1Health.current = 90;
-	e1Health.max = 90;
+        // run a couple searches based on the above entities
+        REQUIRE(ecs.EntitiesWith(Identity{}).size() == 1);
+        REQUIRE(ecs.EntitiesWith(Health{}).size() == 2);
+        REQUIRE(ecs.EntitiesWith().size() == 3);
+    }
 
-	Entity e0 = ecs.CreateEntity();
-	Entity e1 = ecs.CreateEntity();
-	ecs.AddComponent(e0, e0Ident);
-	ecs.AddComponent(e0, e0Health);
+    TEST_CASE("EntitysWith finds 2 entities with Health")
+    {
+        REQUIRE(ecs.EntitiesWith(Health{}).size() == 2);
+    }
 
-	ecs.AddComponent(e1, e1Ident);
-	ecs.AddComponent(e1, e1Health);
+    TEST_CASE("EntitiesWith finds all entities when no components are specified")
+    {
+        REQUIRE(ecs.EntitiesWith().size() == 3);
+    }
 
-	// has Health
-	auto health = ecs.GetComponent(e0, Health());
-	REQUIRE(e0Health.current == health->current);
-	REQUIRE(e0Health.max == health->max);
+    TEST_CASE("EntitiesWith with unregistered component should throw")
+    {        
+        // set up entity 1
+        Entity entity1 = ecs.CreateEntity();
+        Health hp{ 100, 44 };
+        ecs.AddComponent<Health>(entity1, hp);
 
-	ecs.RemoveComponent(e0, Health());
+        // Getting component
+        Health* storedHp = ecs.GetComponent(entity1, Health());
 
-	// does not have health anymore
-	auto noHealth = ecs.GetComponent(e0, Health());
+        REQUIRE(storedHp != nullptr);
+        REQUIRE(storedHp->current == hp.current);
+        REQUIRE(storedHp->max == hp.max);
 
-	REQUIRE(noHealth == nullptr);
+        ecs.RegisterComponent(Health());
+        ecs.RegisterComponent(Identity());
 
-	//still has Identity
-	auto identity = ecs.GetComponent(e0, Identity());
-	REQUIRE(identity->name == "babs2");
+        Entity e = ecs.CreateEntity();
 
-	auto entitiesWithHealth = ecs.EntitiesWith(Health());
-
-	REQUIRE(entitiesWithHealth.size() == 1);
+        // Should throw because AI is not registered
```

Comment:
I wrote this comment but it doesn't make any sense.

"Should throw"
"However, it doesn't"

Then it checks that it DOES throw so it must, maybe delete the comment on ln 174?

Reply:
Oh yeah, thats not relevant anymore, I remember that from us debugging something